### PR TITLE
RHBPMS-4165: BRMS Generates wrong source code for Guided Rules if using 'Formula'

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.jboss.org/browse/RHBPMS-4165

### CODE REVIEW

**Comment by manstis** [comment]
Jenkins please retest this.

**Comment by manstis** [comment]
@etirelli @triceo Hi.. IDK how to get this JIRA into "work in progress".. does it need some flag setting?

**Comment by manstis** [comment]
Jenkins please retest this.

**Comment by manstis** [comment]
**DO NOT MERGE YET - TWEAKING FIX**

**Comment by manstis** [comment]
Jenkins please retest this.

File: drools-workbench-models/drools-workbench-models-commons/src/main/java/org/drools/workbench/models/commons/backend/rule/RuleModelPersistenceHelper.java [hunk 1]
```diff
@@ -206,6 +206,34 @@ static int inferFieldNature( final String dataType,
...
+                if ( inString ) {
+                    inString = false;
+                } else {
+                    inString = true;
+                }
+            } else if ( c == '\\' ) {
+                escape = true;
+            } else {
+                escape = false;
+            }
```
Comment (mariofusco) [inline drools-workbench-models/drools-workbench-models-commons/src/main/java/org/drools/workbench/models/commons/backend/rule/RuleModelPersistenceHelper.java:232]: @manstis I think the if block starting at line 222 could be rewritten in a more compact way like this

```
if ( c == '"' ) {
    inString = !inString;
} else {
    escape = c == '\\';
}
```

For the rest this is a +1 for me. Should I merge it as it is or do you want to make this small improvement first?

**Comment by mariofusco** [comment]
Merged by https://github.com/droolsjbpm/drools/compare/18b6e5a...d5bea55

**Pull request closed without merging**